samples/kfp-user-guide/README.md

```diff
@@ -331,3 +333,12 @@ kubectl describe pipelinerun echo
 #   Normal   Running            <invalid> (x11 over 2s)  pipeline-controller  Tasks Completed: 0, Incomplete: 1, Skipped: 0
 #   Normal   Succeeded          <invalid>                pipeline-controller  Tasks Completed: 1, Skipped: 0
 ```
+
+## Best Practice
+
+### Artifacts and Parameter output files for Tekton
+When developing a Kubeflow pipeline for the Tekton backend, please be aware that the files you produced for artifacts and parameter outputs are carried to a volume mount path then get pushed to S3. It's not recommended to have volume mount on a container's root directory (`/`) because volume mount will overwrite all the files in the container path, including all the system files and binaries. Therefore, we have prohibited the kfp-tekton compiler from putting artifacts and parameter output files in the container's root directory.
+
+Therefore, we recommend placing the output files inside a new directory under root to avoid this problem, such as `/tmp/`. The [condition](/samples/flip-coin/condition.py) example shows how the output files can be stored. Alternatively, you can learn [how to create reusable components](https://www.kubeflow.org/docs/pipelines/sdk/component-development/) in a component.yaml where you can avoid hard coding your output file path.
```

Comment:
```suggestion

Therefore, we have prohibited the kfp-tekton compiler from putting artifacts and parameter output files in the container's root directory. We recommend placing the output files inside a new directory under root to avoid this problem, such as `/tmp/`. The [condition](/samples/flip-coin/condition.py) example shows how the output files can be stored. Alternatively, you can learn [how to create reusable components](https://www.kubeflow.org/docs/pipelines/sdk/component-development/) in a component.yaml where you can avoid hard coding your output file path.
```